mitodl/unified-ecommerce
# Adds initial buildout of APIs and completes some project bootstrapping tasks

# What are the relevant tickets?

Part of https://github.com/mitodl/hq/issues/3158

# Description (What does it do?)

This does some bootstrapping stuff and gets a couple of the core APIs and data models in place, including:
- Exposing both Swagger and Redoc versions of the browseable API
- Pulling in resources - fixtures, tests, base models, etc. - from MITx Online and Open
- Setting up some generic test classes for serializers and viewsets
- Cleaning up some documentation
- Getting the data model and APIs in for IntegratedSystem and Product
- Getting management commands in for creating integrated systems and products

# How can this be tested?

Automated tests should pass. 

- You should be able to log into the Django Admin and should be able to interact with the various data models.
- You should be able to see the browseable API at `/api/schema/swagger-ui/` and `/api/schema/redoc/` and interacting with the API should work.
- You should be able to create a new integrated system record with the `add_system` management command.
- You should be able to perform most CRUD operations with products using the `manage_product` management command.

In addition, the structure of the app should make sense, and tests should also be reasonable. 

Note that, especially for API requests, we're not _really_ doing authentication right now; you'll just need a Django session to be able to get to things. [API auth is a future PR.](https://github.com/mitodl/hq/issues/3166)

# Additional Context

There are a few things in here that I'm sort of stuck on - this uses Django 4.2 and the `ol-django` apps don't work in it quite yet. [I have a PR in progress for this](https://github.com/mitodl/ol-django/pull/129) but I didn't want to wait on it, so things like `TimestampedModel` and a few other bits will get pulled in later.
